src/transformers/pipelines/conversational.py

```diff
@@ -19,137 +19,120 @@ class Conversation:
-
-    # Steps usually performed by the model when generating a response:
-    # 1. Mark the user input as processed (moved to the history)
-    conversation.mark_processed()
-    # 2. Append a mode response
-    conversation.append_response("The Big lebowski.")
-
-    conversation.add_user_input("Is it good?")
+    conversation.add_message({"role": "assistant", "content": "The Big lebowski."})
+    conversation.add_message({"role": "user", "content": "Is it good?"})
     ```"""
 
     def __init__(
-        self, text: str = None, conversation_id: uuid.UUID = None, past_user_inputs=None, generated_responses=None
+        self,
+        messages: Union[str, List[Dict[str, str]]] = None,
+        conversation_id: uuid.UUID = None,
+        past_user_inputs=None,
+        generated_responses=None,
     ):
         if not conversation_id:
             conversation_id = uuid.uuid4()
-        if past_user_inputs is None:
-            past_user_inputs = []
-        if generated_responses is None:
-            generated_responses = []
-
-        self.uuid: uuid.UUID = conversation_id
-        self.past_user_inputs: List[str] = past_user_inputs
-        self.generated_responses: List[str] = generated_responses
-        self.new_user_input: Optional[str] = text
+        if messages is None:
+            messages = []
+        elif isinstance(messages, str):
+            messages = [{"role": "user", "content": messages}]
+
+        # This block deals with the legacy args - new code should just totally
+        # avoid past_user_inputs and generated_responses
+        if generated_responses is not None and past_user_inputs is None:
+            raise ValueError("generated_responses cannot be passed without past_user_inputs!")
+        if past_user_inputs is not None:
+            legacy_messages = []
+            if generated_responses is None:
+                generated_responses = []
+            # We structure it this way instead of using zip() because the lengths may differ by 1
+            for i in range(max([len(past_user_inputs), len(generated_responses)])):
+                if i < len(past_user_inputs):
+                    legacy_messages.append({"role": "user", "content": past_user_inputs[i]})
+                if i < len(generated_responses):
+                    legacy_messages.append({"role": "assistant", "content": generated_responses[i]})
+            messages = legacy_messages + messages
+
+        self.uuid = conversation_id
+        self.messages = messages
 
     def __eq__(self, other):
         if not isinstance(other, Conversation):
             return False
-        if self.uuid == other.uuid:
-            return True
-        return (
-            self.new_user_input == other.new_user_input
-            and self.past_user_inputs == other.past_user_inputs
-            and self.generated_responses == other.generated_responses
-        )
-
-    def add_user_input(self, text: str, overwrite: bool = False):
```

Comment:
We need to keep this for backward compatibility. Deprecate but not remove

Reply:
Added it back!